da/celestia/proto/celestia.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+package celestia;
+
+option go_package = "github.com/celestiaorg/optimint/da/celestia/pb";
+import "tendermint/types/types.proto";
+
+service DataRetriever {
+	rpc RetrieveShares (Request) returns (Shares) {}
+	rpc RetrieveStateRelevantMessages (Request) returns (StateRevelantMessages) {}
+	rpc RetrieveMessages (Request) returns (tendermint.types.Messages) {}
```

Comment:
I would think the Requests to be different per RPC endpoint:

1.  not sure we even need an rpc end-point for raw shares; I think one that returns the already merged back messages is sufficient
2. `RetrieveStateRelevantMessages`: here the req does only need to identify for which block it is; this could be done per block height, data hash,  or header hash (IMO block height or data hash makes the most sense)
3. `RetrieveMessages`: here the combination of namespace ID + data hash (or block height) is sufficient